tolstoy/src/tx_client.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2016 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+use std::collections::BTreeMap;
+use std::collections::btree_map::Entry;
+
+use rusqlite;
+
+use errors::{
+    Result,
+    ErrorKind
+};
+
+use mentat_db::types::{
+    Entid
+};
+
+use mentat_db::{
+    entids,
+    TypedSQLValue
+};
+
+use mentat_core::{
+    TypedValue
+};
+
+#[derive(Debug)]
+pub struct TxPart {
+    pub e: Entid,
+    pub a: i64,
+    pub v: TypedValue,
+    pub added: i32
+}
+
+#[derive(Debug)]
+pub struct Tx {
+    pub tx: Entid,
+    pub tx_instant: TypedValue,
+    pub parts: Vec<TxPart>
+}
+
+struct RawTx {
+    e: Entid,
+    a: i64,
+    v: TypedValue,
+    tx: Entid,
+    added: i32
```

Comment:
Again, reconsider `i32`. If it's coming out of SQLite it's probably an `i64`, but we want a `bool`.